### PR TITLE
[dev] Update RPM to 4.17.0

### PR DESCRIPTION
**Pull request by MateuszMalisz**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

This change updates RPM to 4.17.0. This introduces some major changes (the most prominent in my opinion being [removal of some scripting language support](https://github.com/rpm-software-management/rpm/commit/ba85c95963f9b62f237c0442f6b5aca3e355fa83)

###### Change Log  

- To be updated

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Local normal package build
- Local full toolchain build

### CODE REVIEW

File: SPECS/rpm/rpm.spec [hunk 1]
```diff
@@ -107,31 +107,38 @@ Python3 rpm.
...
+rm -r docs/man/{fr,ja,ko,pl,ru,sk}
+
 make %{?_smp_mflags}
 
 pushd python
 python3 setup.py build
 popd
 
+# Set provided python versions
+sed -i 's/@MAJORVER-PROVIDES-VERSIONS@/2.7,3.6/' %{SOURCE3}
```
Comment (PawelWMS) [inline SPECS/rpm/rpm.spec:140]: Let's use the `%python2_version` and `%python3_version` macros instead, so that they are not hard-coded here.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Looks good, a few more things to take care of:
- Update `cgmanifest.json`.
- Update `toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh` to build the new version of `rpm` (will need another toolchain build to verify, unfortunately).
- Update the source tarball name and hash in `toolkit/scripts/toolchain/container/toolchain-sha256sums`.

**Comment by thcrain-msft** [comment]
I do have a few testing questions here: Have you been able to create a worker chroot archive with this upgrade? Are you able to go through a regular package build with this upgrade? Have you been able to to a full build in our pipelines?

I'm asking because I have a larger concern regarding certain dependencies we take on the build host RPM. I noticed once when trying to build on top of some Fedora version that the toolchain built just fine, but failed to create a worker chroot archive. I was able to track down the problem to a difference in supported databases between the version of RPM on the host and the version of RPM from the toolchain. 

Specifically, [this is the one line ](https://github.com/microsoft/CBL-Mariner/blob/d7b579ebfec459dfc0cf7a6f2a4f169c374f6ba5/toolkit/tools/pkggen/worker/create_worker_chroot.sh#L35 )that was causing the problem. Note that this line is using the build host's RPM to install the toolchain packages into a clean chroot.

Since the build host RPM is used for this, whatever database format that the host RPM is configured to use will be used inside the chroot for package management. This database may happen to match with what our RPM version supports (as it does with our supported host of Ubuntu 18.04),  or it could be different (as it is for recent versions of Fedora). If our chroot has an RPM database which is unsupported by the toolchain version of RPM, this causes problems when the build system attempts to install packages inside the worker chroot during package builds.

So, in summary: I'm asking that you run a full build for this before merging, so that we can check for these types of problem and avoid having to revert this.

File: toolkit/resources/manifests/package/pkggen_core_x86_64.txt [hunk 2]
```diff
@@ -176,12 +176,12 @@ libcap-ng-0.7.9-4.cm2.x86_64.rpm
 libdb-5.3.28-5.cm2.x86_64.rpm
 libdb-devel-5.3.28-5.cm2.x86_64.rpm
 libdb-docs-5.3.28-5.cm2.x86_64.rpm
-rpm-4.14.2.1-4.cm2.x86_64.rpm
```
Comment (anphel31) [inline toolkit/resources/manifests/package/pkggen_core_x86_64.txt:179]: Updating these versions will require you to also compile the new rpm version in the toolchain. Otherwise toolchain builds will break when these files are not found

**Review by anphel31 — CHANGES_REQUESTED** [review]
Please update the toolchain scripts to compile the new rpm version
